### DISCUSSION TITLE
Cannot use GraphQLObjectType "Post" from another module or realm.

### DISCUSSION THREAD

**Discussion by simplecommerce** [codesandbox/codesandbox-client#4958] [General]
Hi, 
I have a specific question, a problem I encounter when I run this app on the client instance.

https://codesandbox.io/s/gifted-forest-6vn7j

I keep getting the error 

```
Cannot use GraphQLObjectType "Post" from another module or realm.

Ensure that there is only one instance of "graphql" in the node_modules
directory. If different versions of "graphql" are the dependencies of other
relied on modules, use "resolutions" to ensure only one version is installed.
```

The problem, is that when I download the zip from the client and install it on my webhosting server, I am able to run the app with no issues.

I was wondering, what could be causing the issue, is it something specific to how your client is bundled? 
The developer of the package I am testing told me that it could be a bundler issue.
But I am not able to confirm and reproduce it on my end.

Any help would be appreciated, thanks!

**Comment by CompuIves** [comment]
Ohh this one is interesting. It seems that one file is importing `definition.mjs`, the other file is importing `definition.js`, which causes it to load two instances of GQL. Seems to me like the resolving of files is inconsistent, or that there is an import that imports with extension, and the other one doesn't. Going to check...

**Reply by CompuIves** [reply]
Okay, it's becoming more clear now. The `graphql` package specifies that it uses `mjs` for its `module` field, but we only load `mjs` as last resort (so we find `js` more important). So if there's another package directly requesting a file from the package, we will prefer `.js`. Going to quickly check the CRA config, maybe they actually do prefer `.mjs` over `.js`...

**Reply by CompuIves** [reply]
Ah yes that's it, Webpack prefers `mjs`. We should change that!

**Reply by simplecommerce** [reply]
Sorry to sound dumb, but what does it mean actually in terms of what I would need to do to fix this issue?
Is it something that I can manually add in my application's configuration settings or is this something the library developer has to include or fix?

**Reply by CompuIves** [reply]
Sorry, thinking out loud! It's an issue on our side, I just created a fix in a PR here: https://github.com/codesandbox/codesandbox-client/pull/4962. I will merge and deploy that in about ~30 minutes.

This doesn't fully fix the issue though, because of the cache we have on existing packages. It should probably work if you add `graphql` to the sandbox as a dependency, and it will certainly work if you update `uniforms-bridge-graphql` to a new version (no content changes needed).

**Reply by simplecommerce** [reply]
Ok so from what I understand, if I am having the same issue in my MeteorJS application, that would mean that I would have to open an issue with the MeteorJS team also I presume?

Thanks for the quick help, truly appreciate it!

**Reply by CompuIves** [reply]
Yep, if MeteorJS is having the same issue they also probably prefer `.js` over `.mjs` in resolving files. But can also be something else, I didn't check the code.
